src/test/java/org/broadinstitute/hellbender/utils/io/IOUtilsUnitTest.java

```diff
@@ -196,6 +196,20 @@ private void innerTestGetPath(String s) throws IOException {
         Assert.assertTrue(size>0);
     }
 
+    @DataProvider
+    public Object[][] absoluteNames() {
+        return new Object[][] {
+                {"/local/example.txt", "/local/example.txt"},
+                {"/local/file://example.txt", "/local/file:/example.txt"},
+                {"file:///local/example.txt", "/local/example.txt"}
+        };
+    }
+
+    @Test(dataProvider = "absoluteNames")
+    public void testGetAbsolutePathName(final String uriString, final String expected) {
+        Assert.assertEquals(IOUtils.getAbsolutePathName(IOUtils.getPath(uriString)), expected);
+    }
+
     @Test
```

Comment:
Can you add a unit test to prove that the temp dir works over NIO?

Reply:
Which kind of test do you need there? Retrieving the property and writing to it? Or running tools using it with NIO (e.g., `MarkDuplicatesGATK`)? Let me know for next round of comments...

Reply:
How about a test that:

1. Declares a dummy `CommandLineProgram` that executes `createTempFile()` and then asserts that it exists in its `doWork()` method

2. Runs the above command line program with `--tmp-dir` set to a jimfs URI. You can create a jimfs like this:

```
        try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
            final Path tmpDirPath = jimfs.getPath("tmp");
```

3. You should also include a test case that runs the dummy `CommandLineProgram` with `--tmp-dir` set to a `file://` URI, since that is a different code path.

You could put the test in a new `CommandLineProgramIntegrationTest` class

Reply:
Which `createTempFile` should be use in the test? There are several implementations (htsjdk, gatk, `Files`, `File`), and all of them have their own peculiarities. The ones that work with a `File` object won't work with this change. Thus, I am not sure how to proceed here - for the tools using the tmpDir this should work because the codebase uses the htsjdk `Path` implementation, but otherwise I am not sure how to get this done well...

On the other hand,  there is a test class already for CLP (`CommandLineProgramUnitTest`) - should I still create the `*IntegrationTest` class or just use that one?